pypika/functions.py

```diff
@@ -303,7 +303,7 @@ def __init__(self, date_part, field, alias=None):
         self.field = field
 
     def get_special_params_sql(self, **kwargs):
-        return "FROM {field}".format(field=self.field,)
+        return "FROM {table}.{field}".format(table=self.field.table, field=self.field)
```

Comment:
the functionality for this is already built into the field class. The issue here is that the **kwargs are not being passed in which would tell it to render the table name. To remedy that we can call `field.get_sql(**kwargs)` directly on the field object here instead of casting it to a string. That should fix the issue as well but in a way that lets the field object determine it's own serialization logic.

Reply:
Ah. I missed point.. My solution is fool. I fix and add new testcase for join with `EXTRACT`. Thanks for reviewing :smile: